src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -14,15 +15,23 @@ internal partial class LocalRewriter
     {
         public override BoundNode VisitConversion(BoundConversion node)
         {
-            var rewrittenType = VisitType(node.Type);
             if (node.ConversionKind == ConversionKind.InterpolatedString)
             {
                 return RewriteInterpolatedStringConversion(node);
             }
 
+            var rewrittenType = VisitType(node.Type);
+            var rewrittenOperand = node.Operand;
+
+            if (rewrittenOperand is BoundStackAllocArrayCreation boundStackAlloc)
```

Comment:
>if [](start = 12, length = 2)

If the conversion is a stack alloc conversion, which you seem to be half-handling here, can you please fully translate the conversion along with its operand? You should be able to discard the conversion and fix up the stackalloc node.